### PR TITLE
refactor!: use `attrs.asdict` as a way to get json data of the object

### PR DESCRIPTION
**Pull request by Damego**
## About

This pull request:
1. Makes `DictSerializerMixin._json` as a property with using `attrs.asdict` to get json data of the object. 
2. Solves all troubles with editing json. You don't need to edit json anymore lol.
3. Removes `ComponentMixin` and json modifing in a lot of objects like `Embed`.
4. Limited using _json property in some places: using walrus operators and objects instead of dicts(wsclient).
5. Removes outdated(since attrs added) code like [here](https://github.com/interactions-py/library/pull/1126/files#diff-9c1dc218ce0d42c712c5591f6797cead2f2b2084810fb6d9f66dd48bf1c112ebL399)

## Checklist

- [x] The ``pre-commit`` code linter has been run over all edited files to ensure the code is linted.
- [x] I've ensured the change(s) work on `3.8.6` and higher.

I've made this pull request: (check all that apply)
  - [ ] For the documentation
  - [ ] To add a new feature
  - [x] As a general enhancement
  - [x] As a refactor of the library/the library's code
  - [ ] To fix an existing bug
  - [ ] To resolve #ISSUENUMBER

This is:
  - [x] A breaking change

### CODE REVIEW

**Comment by Damego** [comment]
Ready for review. Any tests appreciated

**Comment by sillyElvy** [comment]
Do not merge for another few days as this is a major change

**Review by sillyElvy — CHANGES_REQUESTED** [review]
> Optimized code in some places: using walrus operators and objects instead of dicts(wsclient).


Out of the scope of this PR, please remove

**Comment by Damego** [comment]
> > Optimized code in some places: using walrus operators and objects instead of dicts(wsclient).
> 
> Out of the scope of this PR, please remove

Remove changes or just these words?

**Comment by sillyElvy** [comment]
> > > Optimized code in some places: using walrus operators and objects instead of dicts(wsclient).
> > 
> > Out of the scope of this PR, please remove
> 
> Remove changes or just these words?

The changes

**Comment by Damego** [comment]
uhh. It can't be fully reverted because someone previously added [object into json](https://github.com/interactions-py/library/pull/1126/files#diff-9c1dc218ce0d42c712c5591f6797cead2f2b2084810fb6d9f66dd48bf1c112ebL373) but we can't do it anymore and that's why i refactored `__sub_command_context` function.

**Comment by sillyElvy** [comment]
> uhh. It can't be fully reverted because someone previously added [object into json](https://github.com/interactions-py/library/pull/1126/files#diff-9c1dc218ce0d42c712c5591f6797cead2f2b2084810fb6d9f66dd48bf1c112ebL373) but we can't do it anymore and that's why i refactored `__sub_command_context` function.

It sounded like a random change, if it is only _json related it's ok

**Comment by Damego** [comment]
also using _json property few times in one funtion is a bad. It will everytime call `asdict`. Yeah its takes just 1-2 milliseconds but still

**Comment by sillyElvy** [comment]
Yeah

**Comment by Damego** [comment]
> Yeah

So remove these changes(using walrus operator)?

**Comment by sillyElvy** [comment]
> > Yeah
> 
> So remove these changes(using walrus operator)?

If you did the changes while removing the _json it can stay.

**Comment by Damego** [comment]
Draft until #1132 is merged because 
> ... this is a major change

xD

File: interactions/utils/attrs_utils.py
Comment (Damego) [inline interactions/utils/attrs_utils.py:99]:
```suggestion
                return int(value)
```

File: interactions/api/gateway/client.py
Comment (sillyElvy) [inline interactions/api/gateway/client.py:391]: IN the upper part you are doing the check, In the lower part you removed it. why?

File: interactions/client/models/component.py
Comment (sillyElvy) [inline interactions/client/models/component.py]:
```suggestion
    def new(cls, *components: Union[Button, SelectMenu, TextInput]) -> List["ActionRow"]:
```

If you change the method you can fix the rtype aswell

File: interactions/client/models/component.py
Comment (sillyElvy) [inline interactions/client/models/component.py]:
```suggestion
```

Include it and let it error, it produces less confusion than just to let it out

File: interactions/client/models/component.py
Comment (sillyElvy) [inline interactions/client/models/component.py]:
```suggestion
```

File: interactions/client/models/component.py
Comment (sillyElvy) [inline interactions/client/models/component.py]:
```suggestion
```

File: interactions/client/models/component.py
Comment (sillyElvy) [inline interactions/client/models/component.py]:
```suggestion
```

File: interactions/utils/attrs_utils.py
Comment (sillyElvy) [inline interactions/utils/attrs_utils.py]: Snowflakes are technically strings

**Review by sillyElvy — CHANGES_REQUESTED** [review]

File: interactions/api/gateway/client.py
Comment (sillyElvy) [reply interactions/api/gateway/client.py:391]: WAIT NVM IM STUPID

File: interactions/api/gateway/client.py
Comment (sillyElvy) [inline interactions/api/gateway/client.py]: It should assign the resolved object to `option.value` so that it is not the ID of the object if I'm reading it correctly

Comment (Damego) [reply interactions/api/gateway/client.py]: yeah, you're right. I removed that comment

**Comment by sillyElvy** [comment]
Also merge conflicts

**Comment by Damego** [comment]
Resolved

**Comment by mAxYoLo01** [comment]
Tested PR and works on my end, LGTM!

**Review by sillyElvy — APPROVED** [review]
Since the other devs don't seem interested in having a look at this PR I'm approving it

**Pull request merged**
